src/app/models/function.rb

```diff
@@ -6,11 +6,30 @@ class Function < ApplicationRecord
 
   belongs_to :user
 
+  before_validation :function_name
+
   def to_param
     name
   end
 
   def self.from_param(param)
     find_by(name: param)
   end
+
+  private
+
+  def walk_ast(ast)
+    return ast.children.first.to_s if ast.respond_to?('type') && ast.type == :DEFN
```

Comment:
[Q] is this safe navigation operator ok for rubocop? if not then ignore this suggestion
```suggestion
    return ast.children.first.to_s if ast&.type == :DEFN
```

Reply:
The safe navigation operator works only with instance variables not local methods variables.